drivers/flash/Kconfig.nrf

```diff
@@ -27,6 +27,47 @@ config SOC_FLASH_NRF_RADIO_SYNC
 	help
 	  Enable synchronization between flash memory driver and radio.
 
+if HAS_HW_NRF_NVMC_PE
+
+config SOC_FLASH_NRF_PARTIAL_ERASE
+	bool "Nordic nRFx flash driver partial erase"
+	default n
+	depends on HAS_HW_NRF_NVMC_PE
+	help
+	  Enable partial erase feature. Partial erase is performed in time
+	  slices instead of blocking MCU, for the time it is needed to
+	  complete operation over given area.
+	  This allows interrupting flash erase beetween operations
+	  to perform other task by MCU.
+	  This feature may also be used for better syncing flash erase
+	  operations, when compiled with SOC_FLASH_NRF_RADIO_SYNC,
+	  with Bluetooth.
+
+if SOC_FLASH_NRF_PARTIAL_ERASE
+
+# This is workaround for problem with undefined range when SOC does
+# not define SOC_NVMC_MAX_PE_TIME_MS, as it will happen if that SOC
+# does not support partial erase.
+if !HAS_HW_NRF_NVMC_PE
+config SOC_NVMC_MAX_PE_TIME_MS
+	int
+	default 3
+endif # !HAS_HW_NRF_NVMC_PE
```

Comment:
This is quite awful. And definitely not worth the effect, IMHO of course.

Reply:
Removed